packages/workflow/src/records/fhir.ts

```diff
@@ -428,6 +428,61 @@ export function createValidateTask(
   }
 }
 
+export function createArchiveTask(
+  previousTask: SavedTask,
+  practitioner: Practitioner,
+  reason?: string,
+  comment?: string,
+  duplicateTrackingId?: string
+): SavedTask {
+  const extension: Extension[] = [
+    ...previousTask.extension.filter((extension) =>
+      [
+        'http://opencrvs.org/specs/extension/contact-person-phone-number',
+        'http://opencrvs.org/specs/extension/informants-signature',
+        'http://opencrvs.org/specs/extension/contact-person-email'
+      ].includes(extension.url)
+    )
+  ]
+
+  if (duplicateTrackingId)
+    extension.push({
+      url: `http://opencrvs.org/specs/extension/duplicateTrackingId`,
+      valueString: duplicateTrackingId
+    })
+
+  return {
+    resourceType: 'Task',
+    status: 'accepted',
+    intent: 'proposal',
+    code: previousTask.code,
+    focus: previousTask.focus,
+    id: previousTask.id,
+    requester: {
+      agent: { reference: `Practitioner/${practitioner.id}` }
+    },
+    identifier: previousTask.identifier,
+    extension,
+    // Reason example - "duplicate"
+    reason: { text: reason ?? '' },
```

Comment:
I wonder which would be better, keeping the reason with an empty string or not adding in the reason at all in that case